x-pack/plugins/apm/server/routes/metrics/by_agent/serverless/compute_usage.ts

```diff
@@ -0,0 +1,167 @@
+import { GenericMetricsChart } from '../../fetch_and_transform_metrics';
+
+const chartBase: ChartBase = {
+  title: i18n.translate('xpack.apm.agentMetrics.serverless.computeUsage', {
+    defaultMessage: 'Compute usage',
+  }),
+  key: 'compute_usage',
+  type: 'linemark',
+  yUnit: 'number',
+  series: {
+    computeUsage: {
+      title: i18n.translate('xpack.apm.agentMetrics.serverless.computeUsage', {
+        defaultMessage: 'Compute usage',
+      }),
+    },
+  },
+};
+
+/**
+ * To calculate the compute usage we need to multiple the "system.memory.total" by "faas.billed_duration".
+ * But the result of this calculation is in Bytes-milliseconds, as the "system.memory.total" is stored in bytes and the "faas.billed_duration" is stored in milliseconds.
+ * But to calculate the overall cost AWS uses GB-second, so we need to convert the result to this unit.
+ */
+function calculateComputeUsageGBSeconds({
+  faasBilledDuration,
+  totalMemory,
+}: {
+  faasBilledDuration?: number | null;
+  totalMemory?: number | null;
+}) {
+  if (
+    faasBilledDuration === undefined ||
+    faasBilledDuration === null ||
+    totalMemory === undefined ||
+    totalMemory === null
+  ) {
+    return 0;
+  }
+  const bytesMsResult = totalMemory * faasBilledDuration;
+  const GBSeconds = 1024 * 1024 * 1024 * 1000;
+  return bytesMsResult / GBSeconds;
+}
+
+export async function getComputeUsage({
+  environment,
+  kuery,
+  setup,
+  serviceName,
+  faasId,
+  start,
+  end,
+}: {
+  environment: string;
+  kuery: string;
+  setup: Setup;
+  serviceName: string;
+  faasId?: string;
+  start: number;
+  end: number;
+}): Promise<GenericMetricsChart> {
+  const { apmEventClient, config } = setup;
+
+  const aggs = {
+    avgFaasBilledDuration: { avg: { field: FAAS_BILLED_DURATION } },
+    avgTotalMemory: { avg: { field: METRIC_SYSTEM_TOTAL_MEMORY } },
+  };
+
+  const params = {
+    apm: {
+      events: [ProcessorEvent.metric],
+    },
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            { term: { [SERVICE_NAME]: serviceName } },
+            ...rangeQuery(start, end),
+            ...environmentQuery(environment),
+            ...kqlQuery(kuery),
+            ...termQuery(FAAS_ID, faasId),
+            { exists: { field: FAAS_COLDSTART_DURATION } },
+          ],
+        },
+      },
+      aggs: {
+        timeseriesData: {
+          date_histogram: getMetricsDateHistogramParams({
+            start,
+            end,
+            metricsInterval: config.metricsInterval,
+          }),
+          aggs,
+        },
+        ...aggs,
+      },
+    },
+  };
+
+  const { hits, aggregations } = await apmEventClient.search(
+    'get_compute_usage',
+    params
+  );
+  const timeseriesData = aggregations?.timeseriesData;
+
+  return {
+    title: chartBase.title,
+    key: chartBase.key,
+    yUnit: chartBase.yUnit,
+    series:
+      hits.total.value === 0
```

Comment:
maybe you've copied this over from somewhere else, but I don't think using `hits.total.value` for this purpose is a good idea. esp because we are considering setting track_total_hits to false everywhere we're only using aggs. See https://github.com/elastic/kibana/issues/139703

Reply:
I think we want to avoid tracking total hits entirely, because the optimisation might very well not come from the number of hits to track but from the fact that it doesn't have to track any hits at all (and thus can avoid some initial cost). In this case all buckets will be empty so we can check for that.

Reply:
sounds good, so I'll check if the buckets is not empty.